program/object-detection-armnn-tflite/includes/settings.h

```diff
@@ -0,0 +1,365 @@
+        _full_report = get_yes_no(getenv("FULL_REPORT"));
+
+        _m_max_classes_per_detection = std::stoi(alter_str(getenv("MAX_CLASSES_PER_DETECTION"), "1"));
+        _m_max_detections = std::stoi(alter_str(getenv("MAX_DETECTIONS"), getenv("CK_ENV_TENSORFLOW_MODEL_MAX_DETECTIONS")));
+        _m_max_total_detections = std::max(100, _m_max_detections);
+        _m_detections_per_class = std::stoi(alter_str(getenv("DETECTIONS_PER_CLASS"), "100"));
+        _m_num_classes = std::stoi(alter_str(getenv("NUM_CLASSES"), getenv("CK_ENV_TENSORFLOW_MODEL_NUM_CLASSES"))) + _correct_background;
+        _m_nms_score_threshold = std::stof(alter_str(getenv("NMS_SCORE_THRESHOLD"), getenv("CK_ENV_TENSORFLOW_MODEL_NMS_SCORE_THRESHOLD")));
+        _m_nms_iou_threshold = std::stof(alter_str(getenv("NMS_IOU_THRESHOLD"), getenv("CK_ENV_TENSORFLOW_MODEL_NMS_IOU_THRESHOLD")));
+        _m_h_scale = std::stof(alter_str(getenv("H_SCALE"), getenv("CK_ENV_TENSORFLOW_MODEL_SCALE_H")));
+        _m_w_scale = std::stof(alter_str(getenv("W_SCALE"), getenv("CK_ENV_TENSORFLOW_MODEL_SCALE_W")));
+        _m_x_scale = std::stof(alter_str(getenv("X_SCALE"), getenv("CK_ENV_TENSORFLOW_MODEL_SCALE_X")));
+        _m_y_scale = std::stof(alter_str(getenv("Y_SCALE"), getenv("CK_ENV_TENSORFLOW_MODEL_SCALE_Y")));
+
+        _d_boxes = new float [_m_anchors_count * 4]();
+        _d_scores = new float [_m_anchors_count * _m_num_classes];
+
+        _d_scores_sort_buf = new float[_m_max_total_detections + _m_max_classes_per_detection];
+        _d_classes_sort_buf = new int[_m_max_total_detections + _m_max_classes_per_detection];
+        _d_boxes_sort_buf = new int[_m_max_total_detections + _m_max_classes_per_detection];
+        _d_classes_ids_sort_buf = new int[_m_num_classes];
+
+        // Print settings
+        if (_verbose || _full_report) {
+            std::cout << "Graph file: " << _graph_file << std::endl;
+            std::cout << "Image dir: " << _images_dir << std::endl;
+            std::cout << "Image list: " << _images_file << std::endl;
+            std::cout << "Image size: " << _image_size_width << "*" << _image_size_height << std::endl;
+            std::cout << "Image channels: " << _num_channels << std::endl;
+            std::cout << "Result dir: " << _detections_out_dir << std::endl;
+            std::cout << "Batch count: " << _batch_count << std::endl;
+            std::cout << "Batch size: " << _batch_size << std::endl;
+            std::cout << "Normalize: " << _normalize_img << std::endl;
+            std::cout << "Subtract mean: " << _subtract_mean << std::endl;
+            std::cout << "Use Neon: " << _use_neon << std::endl;
+            std::cout << "Use OpenCL: " << _use_opencl << std::endl;
+        }
+
+        // Create results dir if none
+        auto dir = opendir(_detections_out_dir.c_str());
+        if (dir)
+            closedir(dir);
+        else
+            system(("mkdir " + _detections_out_dir).c_str());
```

Comment:
Use a more portable construct to create dir?

Reply:
Fixed